.github/workflows/node-trusted-publish.yml

```diff
@@ -0,0 +1,57 @@
+on:
+  workflow_call:
+    inputs:
+      node-version:
+        required: false
+        type: string
+        default: 24.x
+      package-path:
+        required: false
+        type: string
+        default: './dist/package.json'
+      access:
+        required: false
+        type: string
+        default: 'restricted' # 'public' or 'restricted'
+
+jobs:
+  node-publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write  # Required for OIDC
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js ${{ inputs.node-version }}
+        uses: actions/setup-node@v6
+        with:
+          node-version: ${{ inputs.node-version }}
+          cache: 'npm'
+
+      # Note: Trusted publishing requires npm CLI version 11.5.1 or later.
+      - name: Ensure NPM 11.5.1 or higher
+        run: |
+          CURRENT_NPM_VERSION=$(npm -v)
+          REQUIRED_NPM_VERSION="11.5.1"
+          if [ "$(printf '%s\n' "$REQUIRED_NPM_VERSION" "$CURRENT_NPM_VERSION" | sort -V | head -n1)" != "$REQUIRED_NPM_VERSION" ]; then
+            echo "NPM version $CURRENT_NPM_VERSION is too old. Upgrading to $REQUIRED_NPM_VERSION..."
+            npm install -g npm@$REQUIRED_NPM_VERSION
```

Comment:
Do we need this? Can we just bail instead and expect people to use a good node version? I just wonder if there are any security implications of this...